### PR TITLE
chore: :axe:  Migrate help site from IIS to Apache

### PR DESCRIPTION
**Pull request by darcywong00**
For the help.keyman.com portion of #keymanapp/keyman.com#337

This is the initial pass of converting from web.config to .htaccess files so the help.keyman.com site can run on Linux.

* I left `php_value` intact in the root .htaccess file since that will change on the developer/deployment.

I also haven't setup SSL locally so my dev also has the following commented out:
https://github.com/keymanapp/help.keyman.com/blob/64b174b0f2b1fc5eb9485809994259980f2d5dbe/.htaccess#L8-L13

@jahorton and @ermshiperete have previously run their local site on non-Windows, so maybe they can spot check.

----
I'll be verifying the links with 
```
composer check-links
```

### CODE REVIEW

File: _includes/includes/page-version.php [hunk 1]
```diff
@@ -130,7 +130,7 @@ static private function GetContentVersions($path) {
     static private function GetCurrentVersion($versions) {
       $currentVersion = $versions[0];
 
-      $rootwebconfig = @file_get_contents($_SERVER['DOCUMENT_ROOT'] . '/web.config');
+      $rootwebconfig = @file_get_contents($_SERVER['DOCUMENT_ROOT'] . '/.htaccess');
```
Comment (darcywong00) [inline _includes/includes/page-version.php:133]: Should this be configurable depending on the CI OS?

File: .htaccess [hunk 2]
```diff
@@ -161,9 +174,10 @@ RewriteRule "^(.+)$" "$1.php"
...
+RewriteRule "^(.+)/$" "/_includes/md/mdhost.php?file=$1/index.md"
 
-# Rewrite folder/ to folder/index.php
-RewriteCond "%{DOCUMENT_ROOT}/$1/index.php" -f
-RewriteCond %{DOCUMENT_ROOT}/$1/index.md !-f
-RewriteRule "^(.+)$" "$1/index.php"
+# Rewrite folder/ to folder/index.php - Infinite redirection?
+#RewriteCond "%{DOCUMENT_ROOT}/$1/index.php" -f
+#RewriteCond "%{DOCUMENT_ROOT}/$1/index.md" !-f
+#RewriteRule "^(.+)/$" "$1/index.php"
```
Comment (darcywong00) [inline .htaccess:183]: I had to comment out this block in order for folders like /products/ to display

File: .htaccess [hunk 3]
```diff
@@ -151,8 +164,8 @@ RewriteRule "^(.+[^/])$" "$1" [L]
 RewriteCond "%{DOCUMENT_ROOT}/$1.md" -f
 RewriteRule "^(.+)$" "/_includes/md/mdhost.php?file=$1.md"
 
-RewriteCond "%{DOCUMENT_ROOT}/$1" -f
-RewriteRule "^(.+\.md)$" "/_includes/md/mdhost.php?file=$1"
+RewriteCond "%{DOCUMENT_ROOT}/$1.md" -f
+RewriteRule "^(.+)\.md$" "/_includes/md/mdhost.php?file=$1.md"
```
Comment (darcywong00) [inline .htaccess:168]: For some reason, the original syntax would have some markdown help pages display as raw markdown files.

e.g. /developer/language/reference/begin.md

File: _includes/includes/page-version.php (hunk 1 above)
Comment (darcywong00) [reply _includes/includes/page-version.php]: I made an attempt using PHP_OS_FAMILY

File: .htaccess [hunk 4]
```diff
@@ -29,6 +31,18 @@ RewriteEngine on
...
+
+# /products first
+RedirectMatch "/products/(android|iphone-and-ipad|linux|mac|web|windows)/(?!current-version|version-history|\d+\.\d+)(.*)" "/products/$1/current-version/$2"
+
+# /developer/engine/ for Android and iOS
+RedirectMatch "/developer/engine/(android|iphone-ipad)/(?!current-version|\d+\.\d+)(.*)" "/developer/engine/$1/current-version/$2"
+
+# /developer/engine/web
+RedirectMatch "/developer/engine/web/(?!faq|history|current-version|\d+\.\d+)(.*)" "/developer/engine/web/current-version/$1"
+
```
Comment (darcywong00) [inline .htaccess:45]: I tried consolidating the web.configs from /products/ and /developer/engine

**Comment by darcywong00** [comment]
I'll go ahead and mark this ready for review. 
Issues to be resolved on follow-on PRs, along with 

TODO
* Adjusting the GitHub action to run on Linux

File: developer/core/desktop/.htaccess [hunk 5]
```diff
@@ -0,0 +1,6 @@
+
+# Desktop engine content hasn't changed between 11.0 and 13.0 -> 11.0 
+RedirectMatch "(12|13)\.0(.*)" "/developer/core/desktop/current-version$2"
```
Comment (ermshiperete) [inline developer/core/desktop/.htaccess:3]: It seems strange that we redirect 12.0 and 13.0 to current-version, but not 14.0 nor 15.0...

File: products/desktop/.htaccess [hunk 6]
```diff
@@ -0,0 +1,3 @@
+
+# Redirect desktop to windows
+RedirectMatch "(.*)" "/products/windows/$2"
```
Comment (ermshiperete) [inline products/desktop/.htaccess:3]:
```suggestion
RedirectMatch "(.*)" "/products/windows/$1"
```
?

File: developer/core/desktop/.htaccess (hunk 5 above)
Comment (darcywong00) [reply developer/core/desktop/.htaccess:3]: Good catch. It looks like a bug in the live site:
11.0 --> 11.0
12.0 --> current-version (15.0)
13.0 --> current-version (15.0)
14.0 --> 14.0
15.,0 --> 15.0 (current-version)

I think we should fix the 12.0 and 13.0 redirects on a separate PR

File: .htaccess [hunk 7]
```diff
@@ -29,6 +31,18 @@ RewriteEngine on
 # RewriteCond "%{DOCUMENT_ROOT}/$1/XX.0" -d
 # RewriteRule "^(.+)/current-version(/.*|$)" "$1/XX.0$2"
 
+
+# Redirect unversioned paths to current-version/ path
+
+# /products first
+RedirectMatch "/products/(android|iphone-and-ipad|linux|mac|web|windows)/(?!current-version|version-history|\d+\.\d+)(.*)" "/products/$1/current-version/$2"
```
[the remark this answers is not in the document]
Comment (darcywong00) [reply .htaccess:38]: As far as I can tell from my local testing, it seems like the negative regex `(?!current-version...)` doesn't return a grouping so
help.keyman.com.local/products/android/context redirects to
help.keyman.com.local/products/android/15.0/context/

File: developer/core/desktop/.htaccess (hunk 5 above)
Comment (darcywong00) [reply developer/core/desktop/.htaccess]: I fixed this .htaccess file in 9732f34f1b03c0bf992ea52e131f3aa065cc621d

**Review by ermshiperete — APPROVED** [review]

**Review by ermshiperete — APPROVED** [review]

**Comment by darcywong00** [comment]
Also TODO for follow-on PR
Linux has case-sensitive filenames, so some resources currently `.PNG` will need to be renamed. Affects numerous `/products/windows/` images from 7.0 to 11.0.

**Comment by darcywong00** [comment]
I haven't completed integration of getting the site to work in Docker. So I'll merge this into staging to pick up another time.

**Pull request merged**